nifi-nar-bundles/nifi-standard-bundle/nifi-jolt-transform-json-ui/src/main/java/org/apache/nifi/web/standard/api/transformjson/TransformJSONResource.java

```diff
@@ -84,6 +84,36 @@ public Response executeSpec(JoltSpecificationDTO specificationDTO) {
 
     }
 
+    protected Transform getTransformation(JoltSpecificationDTO specificationDTO) throws Exception{
+
+        Object specJson = getSpecificationJsonObject(specificationDTO.getSpecification());
+        String transformName = specificationDTO.getTransform();
+        String modules = specificationDTO.getModules();
+
+        ClassLoader customClassLoader = null;
+        Transform transform ;
 
+        if(modules != null && !modules.isEmpty()){
+            customClassLoader = ClassLoaderUtils.getCustomClassLoader(specificationDTO.getModules(),this.getClass().getClassLoader(), getJarFilenameFilter());
+        }
+
+        if(transformName.equals("jolt-transform-custom")) {
+            transform = TransformFactory.getCustomTransform(customClassLoader,specificationDTO.getCustomClass(), specJson);
```

Comment:
Is it possible that customClassLoader would be null here? I see you check for that in the "else" clause but not the "if" clause. Even if the UI / customValidators prevent this kind of thing, it might still be a good thing to add. If there is a way to directly make the API call without a value for "modules", then this could cause an NPE and a unit test could show that the added null check works :)